### DISCUSSION TITLE
*** Error in `lmp': free(): invalid size: 0x00002b478c4dd040 ***

### DISCUSSION THREAD

**Discussion by lammps-vasp** [deepmodeling/deepmd-kit#488] [Q&A]
2021-04-08 11:15:55.436167: W tensorflow/core/framework/op_kernel.cc:1767] OP_REQUIRES failed at concat_op.cc:161 : Resource exhausted: OOM when allocating tensor with shape[173107136,50] and type double on /job:localhost/replica:0/task:0/device:CPU:0 by allocator cpu
2021-04-08 11:15:55.483135: W tensorflow/core/framework/op_kernel.cc:1767] OP_REQUIRES failed at matmul_op.cc:481 : Resource exhausted: OOM when allocating tensor with shape[173107136,50] and type double on /job:localhost/replica:0/task:0/device:CPU:0 by allocator cpu
*** Error in `lmp': free(): invalid size: 0x00002b478c4dd040 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x81499)[0x2b445499a499]
/public/home/users/caep003/DP_ZHANG/deepmd-kit/bin/../lib/libtensorflow_framework.so.2(+0xe39421)[0x2b4452424421]
/public/home/users/caep003/DP_ZHANG/deepmd-kit/bin/../lib/libtensorflow_framework.so.2(_ZN10tensorflow6TensorD1Ev+0x46)[0x2b4452426886]
/public/home/users/caep003/DP_ZHANG/deepmd-kit/bin/../lib/libtensorflow_framework.so.2(+0xf1e8c2)[0x2b44525098c2]
/public/home/users/caep003/DP_ZHANG/deepmd-kit/bin/../lib/libtensorflow_framework.so.2(+0xfcbd81)[0x2b44525b6d81]
/public/home/users/caep003/DP_ZHANG/deepmd-kit/bin/../lib/libtensorflow_framework.so.2(_ZNSt17_Function_handlerIFvvEZN10tensorflow6thread16EigenEnvironment12CreateThreadESt8functionIS0_EEUlvE_E9_M_invokeERKSt9_Any_data+0x57)[0x2b44525b2e47]
/public/home/users/caep003/DP_ZHANG/deepmd-kit/bin/../lib/libtensorflow_framework.so.2(+0xfb915c)[0x2b44525a415c]
/lib64/libpthread.so.0(+0x7e25)[0x2b4454cede25]
/lib64/libc.so.6(clone+0x6d)[0x2b4454a17bad]
======= Memory map: ========
2b4444509000-2b444452b000 r-xp 00000000 08:03 794042 /usr/lib64/ld-2.17.so
2b444452b000-2b4444540000 rw-p 00000000 00:00 0
2b444455f000-2b4444562000 rw-p 00000000 00:00 0
2b4444562000-2b4444604000 r--p 00000000 00:27 12741564168 /public/home/users/caep003/DP_ZHANG/deepmd-kit/lib/libstdc++.so.6.0.26
2b4444604000-2b4444683000 r-xp 000a2000 00:27 12741564168 /public/home/users/caep003/DP_ZHANG/deepmd-kit/lib/libstdc++.so.6.0.26
2b4444683000-2b44446c4000 r--p 00121000 00:27 12741564168 /public/home/users/caep003/DP_ZHANG/deepmd-kit/lib/libstdc++.so.6.0.26
2b44446c4000-2b44446cf000 r--p 00161000 00:27 12741564168 /public/home/users/caep003/DP_ZHANG/deepmd-kit/lib/libstdc++.so.6.0.26
2b44446cf000-2b44446d3000 rw-p 0016c000 00:27 12741564168 /public/home/users/caep003/DP_ZHANG/deepmd-kit/lib/libstdc++.so.6.0.26

**Comment by AnguseZhang** [comment]
1. Please provide necessary information including the version of DeePMD-kit  and installation way, input file, running commands, as DETAILED as possible, to help locate and reproduce your problem.

2. It seems something like exhausted resources. What's the number of your system?

**Comment by lammps-vasp** [comment]
offline packagesV1.3.2
CentOS Linux release 7.5（64位）






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "deepmodeling/deepmd-kit"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月8日(星期四) 下午3:19
***@***.***&gt;;
抄送:&nbsp;"Zhifeng ***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/deepmd-kit] *** Error in `lmp': free(): invalid size: 0x00002b478c4dd040 *** (#488)





  
 
Please provide necessary information including the version of DeePMD-kit  and installation way, input file, running commands, as DETAILED as possible, to help locate and reproduce your problem.
 
 
 
It seems something like exhausted resources. What's the number of your system?
 
  
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.


从QQ邮箱发来的超大附件

U-Mo.lmp (266.97M, 2021年05月08日 16:01 到期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?t=exs_ftn_download&k=0b30366514b663dd743ddf3d4234011d05530e53550607511b0054575519070054531b045004511f54540f065151030504080003643c33671b7d594b085943320b&code=606ed432

**Reply by amcadmus** [reply]
Could you please try v1.3.3? we have fixed several bugs related to memory in v1.3.3.

**Comment by lammps-vasp** [comment]
OK THANKS


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "deepmodeling/deepmd-kit"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月8日(星期四) 下午4:56
***@***.***&gt;;
抄送:&nbsp;"Zhifeng ***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/deepmd-kit] *** Error in `lmp': free(): invalid size: 0x00002b478c4dd040 *** (#488)





 
Could you please try v1.3.3? we have fixed several bugs related to memory in v1.3.3.

**Comment by AnguseZhang** [comment]
The number of atoms in your system (3456000) is too large.  For current DeePMD-kit version, one GPU card can simulate about 10k to 30k atoms. For model compression version (in on-going development),  one GPU card can simulate about 100k to 300k atoms.  See https://deepmd.readthedocs.io/en/api/use-deepmd-kit.html#compress-a-model.  If you want to simulate more atoms, you should use multiple GPU cards.

**Comment by lammps-vasp** [comment]
I would like to ask whether the frozen model can be converted into a normal format and can be recognized by lammps. If so, how to ensure the invariance of data in the transformation process and whether there are specific conversion methods.


Looking forward to your reply! 




------------------ 原始邮件 ------------------
发件人:                                                                                                                        "deepmodeling/deepmd-kit"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月8日(星期四) 下午4:56
***@***.***&gt;;
抄送:&nbsp;"Zhifeng ***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/deepmd-kit] *** Error in `lmp': free(): invalid size: 0x00002b478c4dd040 *** (#488)





 
Could you please try v1.3.3? we have fixed several bugs related to memory in v1.3.3.
